ducktape/cluster/json.py

```diff
@@ -23,11 +23,14 @@ class JsonCluster(Cluster):
     An implementation of Cluster that uses static settings specified in a cluster file.
     """
 
-    def __init__(self, cluster_json=None):
+    def __init__(self, cluster_json=None, *args, **kwargs):
         super(JsonCluster, self).__init__()
         if cluster_json is None:
-            cluster_json_path = os.path.abspath(os.path.join(os.getcwd(), "cluster.json"))
-            cluster_json = json.load(open(cluster_json_path))
+            # This is a directly instantiation of JsonCluster rather than from a subclass
+            cluster_file = kwargs.get("cluster_file")
```

Comment:
By its current implementation, cluster_file is always provided in kwargs. If its value is not specified in command line, kwargs will have ("cluster_file", None) in it. Then `kwargs.get("cluster_file", ConsoleConfig.CLUSTER_FILE)` will return `None`, which doesn't work for us.

We can replace `session_context.cluster = cluster_class(cluster_file = args.cluster_file)` will the following code in main.py. But it simply moves the extra code from one place to another... what do you think?

```
if (args.cluster_file is None):
  session_context.cluster = cluster_class()
else:
  session_context.cluster = cluster_class(cluster_file = args.cluster_file)
```